Automattic/wp-calypso
# Domain Forwarding: Add subdomain support

Related to https://github.com/Automattic/wp-calypso/issues/81236 
Follow-up of 3512-gh-Automattic/dotcom-forge

## Proposed Changes

* We now can set a subdomain redirect on Domain Forwarding:

Mobile:

<img width="393" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1044309/2dff8318-b7c0-448c-aceb-65da224e8e53">


Desktop:

<img width="760" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1044309/c9d9d39e-f98a-4a26-9587-7851f5a0b6fa">

Disabled state:

<img width="690" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1044309/647f26b1-a01d-44a0-b555-6d272d90d489">


## Testing Instructions



- Create a new subdomain redirect using the Domain Forwarding feature
- Ensure it's redirecting as expected
- Modify it for using domain redirect instead and ensure the old subdomain redirect won't redirect
- After using a domain redirect, ensure it works as expected
- Ensure disabled states and error messages work as expected
- Try to break it :)

## Pre-merge Checklist
